### ISSUE TITLE
Board selector should be usable with the keyboard

### ISSUE THREAD

**Issue by AlbyIanna** [arduino/arduino-ide#1690]
### Describe the problem

The Board Selector is only partially usable via the keyboard. 

### To reproduce

Using the keyboard only:
- try to focus on the board selector
- open it with either ENTER or SPACEBAR
- close it with ESCAPE
- select all the items in the dropdown
You won't be able to do everything 🐛 

When the *Select Other Board and Port* dialog is open, it's not possible to select a board or a port from the lists only using a keyboard.

### Expected behavior

- focusing on the board selector should show a focus outline
- when the board selector is focused it should be openable by pressing ENTER or SPACEBAR
- when the board selector is open, it should be possible to close it pressing ESCAPE
- selecting an item in the board selector and pressing ENTER or SPACEBAR should trigger the action

When the *Select Other Board and Port* dialog is open, it should be possible to select a board a port only using a keyboard.

### Arduino IDE version

any

### Operating system

N/A

### Operating system version

any

### Issue checklist

- [X] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-ide/issues?q=)
- [X] I verified the problem still occurs when using the latest [nightly build](https://www.arduino.cc/en/software#nightly-builds)
- [X] My report contains all necessary details
